src/core/utils/xmlParse.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * @prettier
+ */
+
+const traverseXML = (node, result = {}) => {
+  if (node.nodeType === 1) {
+    // ELEMENT_NODE - parameter
+    const name = node.nodeName
+
+    if (node.nodeName === "parsererror") {
+      throw new Error("Parameter string value must be valid XML")
+    }
+
+    const childNodes = Array.from(node.childNodes)
+
+    if (childNodes.length === 1) {
+      // only one child node - value of the parameter
+      result[name] = traverseXML(childNodes[0])
+    } else if (childNodes.length > 0) {
+      // more than one child node - the parameter is an object or an array
+      const traversedChildren = childNodes.map((child) => traverseXML(child))
+      result[name] = traversedChildren.reduce((paramObj, child) => {
+        Object.entries(child).forEach(([key, value]) => {
```

Comment:
As I mentioned in a comment

> We might need to get rid of nodes with only whitespace characters

Perhaps we could just add `if (typeof child !== "string")` here, as `traversedChildren` will look something like this:
```
[
    "\n\t\t",
    {
        "id": "1"
    },
    "\n\t\t",
    {
        "name": "Dogs"
    },
    "\n\t"
]
```
It seems like the actual parameters should be `{ key: value }` objects.